pkg/storage-init/storage-init.sh

```diff
@@ -3,6 +3,88 @@
 # Copyright (c) 2018 Zededa, Inc.
 # SPDX-License-Identifier: Apache-2.0
 
+zfs_set_parameter() {
+    parameter="$1"
+    value="$2"
+
+    echo "${value}" > /sys/module/zfs/parameters/"${parameter}"
+}
+
+zfs_set_arc_limits() {
+    # NOOP if persist is not ZFS
+    if ! read -r persist_fs < /run/eve.persist_type; then
+       echo "Can not determine persist filesystem type"
+       return 1
+    fi
+
+    if [ "${persist_fs}" != "zfs" ]; then
+       return 0
+    fi
+
+    # Constants
+    zfs_arc_min="$(echo "256*1024*1024" | bc)"
+
+    # can't go below 384 MiB
+    zfs_arc_max_minimum="$(echo "384*1024*1024" | bc)"
+
+    if ! pool_size_bytes="$(chroot /hostfs zpool list -p -H -o size persist)"; then
+       echo "Failed to get pool size"
+       return 1
+    fi
+
+    bc_script="
+    metadata_estimate=(${pool_size_bytes}*0.3/100);
+    arc_max=(${zfs_arc_min} + metadata_estimate);
+    if (arc_max < ${zfs_arc_max_minimum})
+       arc_max = ${zfs_arc_max_minimum}
+    ;
+    arc_max
+    "
+
+    zfs_arc_max="$(echo "${bc_script}" | bc)"
```

Comment:
Should we check for memory on host before setting this option?

Reply:
Good question @petr-zededa.
Don't know if we will document somewhere that ZFS is only supported if the host has at least N GB.
If not we'd actually need to test on 2G, 4G, etc systems and see how things work.
It it doesn't make sense to scale down the arc for such small systems then it would make sense to add a comment in this code that it has been shown to work with at least N GB of memory.

Reply:
Memory check isn't needed, because zfs_arc_max is limiting the maximum size of the ARC. And once we here, we already using zfs. While we probably should have installed it on this node if there is too little RAM.

So it is a thing for anther PR. I would say we should refuse enabling ZFS on the node where 
```
256 MB + 0.3% of total storage > 80% of available RAM
```

Another action item coming from this comment - we should reduce the memory available for VMs by `256 MB + 0.3% of total storage`, otherwise OOM killer might occur on heavy storage workloads.